packages/history-service/src/create-root-location-transformer.ts

```diff
@@ -128,6 +137,29 @@ export function createRootLocationTransformer(
       return getConsumerPath(consumerPaths, consumerUid);
     },
 
+    createLocation(
+      ...consumerLocations: ConsumerLocation[]
+    ): history.Location {
+      if (options.primaryConsumerUid && !consumerLocations.some((consumerLocation) => consumerLocation.historyKey===options.primaryConsumerUid)) {
+        throw new Error('primary consumer is mandatory')
```

Comment:
When writing the error messages, please follow the style that prevails in this project. The first letter is capitalized and we have punctuation.

Reply:
And please add a test for this error, we expect your code to have 100% test coverage.